framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -745,7 +746,7 @@ MeshDiagnosticsGenerator::checkNonConformalMeshFromAdaptivity(
              fine_elements.size() != 3)
       continue;
     else if ((elem_type == TET4 || elem_type == TET10 || elem_type == TET14) &&
-             (fine_elements.size() % 4 != 0))
+             (fine_elements.size() % 2 != 0))
```

Comment:
so we should have a test rn for the case where 4 refined tets share a node on the edge of a coarsened one, next to a coarsened element

can you please add a test for the 6n case?

but yeah this is a good catch

Reply:
We had tests where the number of refined tets meeting at a node was a multiple of 4, yes, but that already included cases with 12 tets (6 tets each from each of two parent tets). I've added a test which forces the choice of a different diagonal when refining in such a way where we can get 10 tets (6 tets from one parent, 4 tets from the second parent). You can confirm this new test fails without the patch. :)

Reply:
new test looks great. I can confirm we catch more now